### DISCUSSION TITLE
Better view of originals folder

### DISCUSSION THREAD

**Discussion by DD8372812** [Radiokot/photoprism-android-client#155] [Ideas]
Hello,

I have a suggestion for better browsing and viewing photos saved in originals folder.

Currently all folders found in the originals folder are listed one by one in menu folder of the app. This is no workable solution. In my case there are 700 folder in that view.

All my photos are stored within "Originals" in hierarchical order e.g.:

2020
- Event 1
-- day 1
- Event 2
2021
- Event 3
-- day 2
-- day 3
- Event 4

It would be nice if the folder structure of the originals folders would be shown one-to-one in the menu "folder" of the app. So you can browse your photos by your own created order.

**Comment by Radiokot** [comment]
Hi. I'm currently working on sort options for folders. There will be "Name" and "Name, descending" options sorting folders by their path. So if you have:
```
/2021
  - 01
    - 05
  - 02
    - 01
/2022
```
Then with the "Name" order you'll see
`/2021/01/05`, `/2021/02/01`, `/2022`

**Comment by DD8372812** [comment]
Hello Oleg,

great to read that.

Would all the folder still be shown in the view or would it be possible to browse the folder structure? Hundreds of folder to scroll is the problem. I would suggest that when you open the menu "folder" only the years-folder will be shown. If you then go to e.g. 2021 only 01 and 02 will be shown and so on.

**Comment by Radiokot** [comment]
Maybe I'll add nested folders in the future. For now, you can use search at the top or the fast scroll handle at the right to quickly jump to the needed folder:
![Screen_Recording_20240924_141656_PhotoPrismGallery-ezgif com-crop](https://github.com/user-attachments/assets/255b1627-a7d3-4fd1-86f2-2f3684ce2ee2)

**Comment by Radiokot** [comment]
Now it is possible to change folder sorting
https://github.com/Radiokot/photoprism-android-client/releases/tag/1.31.0(46)

**Comment by teodorch85** [comment]
Just came to say that I would like nested folders too!

I can't remember all folder names and having almost 20 years of photos in format like "Year/MM - FOLDERNAME/"  it's really difficult to find by scrolling the correct one!

**Reply by Radiokot** [reply]
Hey Theodor,

Until this is implemented, you can navigate your big list of folders easier if you sort it by name and then move your finger over the right edge of the screen – there's a fast scroll handle there. It also works on Calendar and Albums screens.

**Reply by teodorch85** [reply]
Thanks for the tip, but the list of folders is huge! 
Best bet right now is to use search and try to remember the name of the folder.

<img width="319" height="88" alt="image" src="https://github.com/user-attachments/assets/818ed7e4-cf2b-4030-baf8-a8db5a21cbb5" />

Edit:
I see I can search for the year at least so the number of folders is smaller and try to find the correct one.
